crates/biome_js_parser/src/lexer/mod.rs

```diff
@@ -1727,6 +1738,82 @@ impl<'src> JsLexer<'src> {
         }
     }
 
+    /// Attempts to lex a Glimmer template block `<template>...</template>`.
+    /// Returns Some(GLIMMER_TEMPLATE) if a template was found, None otherwise.
+    fn try_lex_glimmer_template(&mut self) -> Option<JsSyntaxKind> {
+        // We're currently at '<', check if followed by 'template'
+        // Peek ahead without consuming
+        let start = self.position;
+
+        // Check for "<template"
+        if self.peek_byte() != Some(b't') {
+            return None;
+        }
+
+        // Check if we have enough bytes for "<template>"
+        let template_bytes = b"template>";
+        let end_pos = self.position + 1 + template_bytes.len();
+        if end_pos > self.source.len() {
+            return None;
+        }
+        if &self.source.as_bytes()[self.position + 1..end_pos] != template_bytes {
+            return None;
+        }
+
+        // Verify that 'template' is followed by '>' or whitespace (not part of identifier)
+        match self.byte_at(9) {
+            // Position after "<template"
+            Some(b'>' | b' ' | b'\t' | b'\n' | b'\r') => {
+                // This is indeed "<template"
+            }
+            _ => return None, // Not a template tag
+        }
+
+        // Consume "<template"
+        self.advance(9);
+
+        // Skip any attributes/whitespace until '>'
+        while let Some(chr) = self.current_byte() {
+            if chr == b'>' {
+                self.next_byte(); // consume '>'
```

Comment:
nit: `next_byte` returns the next byte in addition to advancing the lexer, so you could probably simplify this loop even further